src/commands/commerce/extension/unmap.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Org, SfdxError } from '@salesforce/core';
+import { QueryResult } from '@mshanemc/plugin-helpers/dist/typeDefs';
+import { forceDataSoql, forceDataRecordDelete } from '../../../lib/utils/sfdx/forceDataSoql';
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+import { Result } from '../../../lib/utils/jsonUtils';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:unmap`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class UnMapExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+    public static description = msgs.getMessage('extension.unmap.cmdDescription');
+    public static example = [
+        `sfdx ${CMD} --registered-extension-name test-extension-name --store-name test-store-name `,
+        `sfdx ${CMD} --registered-extension-name test-extension-name --store-id test-store-id `,
+    ];
+    public static flagsConfig = {
+        'registered-extension-name': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.unmap.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
+            char: 'n',
+            description: msgs.getMessage('extension.unmap.StoreNameFlagDescription'),
+        }),
+        'store-id': flags.string({
+            char: 'i',
+            description: msgs.getMessage('extension.unmap.storeId'),
+        }),
+    };
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<void> {
+        this.unmapRecord(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public unmapRecord(extensionName: string, storeName: string, storeId: string, userName: string): void {
+        if (storeName === undefined && storeId === undefined) {
```

Comment:
Don't we want 'or' here? Isn't storeName or storeId being undefined enough to throw an error?

Reply:
With this command, the user only has to pass one or the other. So either storeName could be undefined or storeId could be undefined, but never both, which is when this error will be thrown.